src/ui/public/index_patterns/index_patterns.js

```diff
@@ -41,6 +41,7 @@ export function IndexPatternsProvider(Notifier, Private) {
   self.cache = patternCache;
   self.getIds = getProvider('id');
   self.getTitles = getProvider('attributes.title');
+  self.getIndexPatterns = getProvider();
```

Comment:
I don't remember what that id-comment was about...

I doubt it's uncommon for users to have lots of index patterns, and index patterns that have tons of fields. If we don't need to load up all of the index pattern objects I don't think we should, but that would make this a lot easier.